tests/ui/trivial_map_over_range.stderr

```diff
@@ -0,0 +1,58 @@
+error: map of a trivial closure (not dependent on parameter) over a range
+  --> tests/ui/trivial_map_over_range.rs:14:5
+   |
+LL |     (0..10).map(|_| do_something());
+   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: use: `std::iter::repeat_with(|| do_something()).take(10)`
+   |
+   = note: `-D clippy::trivial-map-over-range` implied by `-D warnings`
+   = help: to override `-D warnings` add `#[allow(clippy::trivial_map_over_range)]`
+
+error: map of a trivial closure (not dependent on parameter) over a range
+  --> tests/ui/trivial_map_over_range.rs:15:5
+   |
+LL |     (3..10).map(|_| do_something());
+   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: use: `std::iter::repeat_with(|| do_something()).take(7)`
+
+error: map of a trivial closure (not dependent on parameter) over a range
+  --> tests/ui/trivial_map_over_range.rs:16:5
+   |
+LL |     (0..10).map(|_| 3);
+   |     ^^^^^^^^^^^^^^^^^^ help: use: `std::iter::repeat_with(|| 3).take(10)`
```

Comment:
My hope for this was that it would get picked up by another lint looking for eager evaluable closures in `repeat_with`. I'm not sure what clippy's thoughts are regarding multi-step corrections: I'd believe that the lint should do as much as it can (i.e. suggest collapsing to `repeat` instead of `repeat_with`.